test/e2e_env/multizone/meshservice/connectivity.go

```diff
@@ -264,6 +275,29 @@ spec:
 		}),
 	)
 
+	type failureCase struct {
+		path         string
+		responseCode int
+	}
+	DescribeTable("Gateway in Kubernetes with incorrect routes",
+		func(given failureCase) {
+			Eventually(func(g Gomega) {
+				response, err := client.CollectFailure(
+					multizone.KubeZone1, "demo-client",
+					fmt.Sprintf("http://edge-gateway-ms.%s:8080/%s", namespace, given.path),
+					client.FromKubernetesPod(clientNamespace, "demo-client"),
+				)
+
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(response.ResponseCode).To(Equal(given.responseCode))
+			}, "30s", "1s").Should(Succeed())
+		},
+		Entry("no matching backend should return 500", failureCase{
+			path:         "no-matching-backend",
+			responseCode: 500,
```

Comment:
why 500? shouldn't it be 404? 🤔 or at least 503...

Reply:
That's existing behavior that gateway API prescribes https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.HTTPRouteRule